application/src/main/java/org/thingsboard/server/actors/app/AppActor.java

```diff
@@ -181,31 +183,37 @@ private void onComponentLifecycleMsg(ComponentLifecycleMsg msg) {
 
     private void onToDeviceActorMsg(TenantAwareMsg msg, boolean priority) {
         if (!deletedTenants.contains(msg.getTenantId())) {
-            TbActorRef tenantActor = getOrCreateTenantActor(msg.getTenantId());
-            if (priority) {
-                tenantActor.tellWithHighPriority(msg);
-            } else {
-                tenantActor.tell(msg);
-            }
+            getOrCreateTenantActor(msg.getTenantId()).ifPresent(tenantActor -> {
+                if (priority) {
+                    tenantActor.tellWithHighPriority(msg);
+                } else {
+                    tenantActor.tell(msg);
+                }
+            });
         } else {
             if (msg instanceof TransportToDeviceActorMsgWrapper) {
                 ((TransportToDeviceActorMsgWrapper) msg).getCallback().onSuccess();
             }
         }
     }
 
-    private TbActorRef getOrCreateTenantActor(TenantId tenantId) {
-        return ctx.getOrCreateChildActor(new TbEntityActorId(tenantId),
-                () -> DefaultActorService.TENANT_DISPATCHER_NAME,
-                () -> new TenantActor.ActorCreator(systemContext, tenantId));
+    private Optional<TbActorRef> getOrCreateTenantActor(TenantId tenantId) {
+        if (systemContext.getServiceInfoProvider().isService(ServiceType.TB_CORE) ||
```

Comment:
I suggest taking this if into the lambda and passing it to ctx.getOrCreateChildActor. Because if the actor is there, then there is no point in checking all these conditions, it is enough to just return it.